charts/thanos/templates/_helpers.tpl

```diff
@@ -135,7 +135,7 @@ The config reloader image to use
 
 {{/* Check if we have query ingress */}}
 {{- define "thanos.queryIngress.enabled" -}}
-  {{- or (and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled) .Values.query.ingress.enabled -}}
+  {{- and .Values.query.enabled (or (and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled) .Values.query.ingress.enabled) -}}
```

Comment:
```suggestion
  {{- or (and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled) (and .Values.query.enabled .Values.query.ingress.enabled) -}}
```

This would be a cleaner pattern.